spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/AggregatePath.java

```diff
@@ -338,4 +429,102 @@ static ColumnInfo of(AggregatePath path) {
 			return new ColumnInfo(columnName, AggregatePathTableUtils.prefixWithTableAlias(path, columnName));
 		}
 	}
+
+	/**
+	 * A group of {@link ColumnInfo} values referenced by there respective {@link AggregatePath}. This is relevant for
+	 * composite ids and references to such ids.
+	 **/
+	class ColumnInfos {
+
+		private final AggregatePath basePath;
+		private final Map<AggregatePath, ColumnInfo> columnInfos;
+
+		/**
+		 * @param basePath The path on which all other paths in the other argument are based on. For the typical case of a
+		 *          composite id, this would be the path to the composite ids.
+		 * @param columnInfos A map, mapping {@literal AggregatePath} instances to the respective {@literal ColumnInfo}
+		 */
+		private ColumnInfos(AggregatePath basePath, Map<AggregatePath, ColumnInfo> columnInfos) {
+
+			this.basePath = basePath;
+			this.columnInfos = columnInfos;
+		}
+
+		public static ColumnInfos empty(AggregatePath base) {
+			return new ColumnInfos(base, new HashMap<>());
+		}
+
+		public ColumnInfo unique() {
+
+			Collection<ColumnInfo> values = columnInfos.values();
+			Assert.state(values.size() == 1, "ColumnInfo is not unique");
+			return values.iterator().next();
+		}
+
+		public ColumnInfo any() {
+
+			Collection<ColumnInfo> values = columnInfos.values();
+			return values.iterator().next();
+		}
+
+		public boolean isEmpty() {
+			return columnInfos.isEmpty();
+		}
+
+		public <T> List<T> toList(Function<ColumnInfo, T> mapper) {
+			return columnInfos.values().stream().map(mapper).toList();
+		}
+
+		public void forEach(BiConsumer<AggregatePath, ColumnInfo> consumer) {
+			columnInfos.forEach(consumer);
+		}
+
+		public <T> T any(BiFunction<AggregatePath, ColumnInfo, T> consumer) {
+
+			Map.Entry<AggregatePath, ColumnInfo> any = columnInfos.entrySet().iterator().next();
+			return consumer.apply(any.getKey(), any.getValue());
+		}
+
+		public ColumnInfo get(AggregatePath path) {
+			return columnInfos.get(path);
+		}
+
+		public AggregatePath fullPath(AggregatePath ap) {
+			return basePath.append(ap);
+		}
+
+		public int size() {
+			return columnInfos.size();
+		}
+	}
+
+	class ColumInfosBuilder {
```

Comment:
this class is becoming `public` API when being defined in an interface. Not sure that's intended.

Reply:
Nope, not intended. Moved to package scope top level class